File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TextureAnimationHandler.java
/**
 * 
 */
package com.abubusoft.xenon.animations;

/**
 * @author Francesco Benincasa
 * 
 */
public class TextureAnimationHandler extends AnimationHandler<TextureKeyFrame> {

	@Override
	protected TextureKeyFrame value(TextureKeyFrame current, long enlapsedTime, TextureKeyFrame next) {
		// questa è un'animazione di tipo discrete, quindi viene preso sempre il valore del frame corente
		return current;
	}

	@Override
	protected TextureKeyFrame buildFrame() {		
		return new TextureKeyFrame();
	}

}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TextureAnimationHandler.kt
/**
 *
 */
package com.abubusoft.xenon.animations

/**
 * @author Francesco Benincasa
 */
class TextureAnimationHandler : AnimationHandler<TextureKeyFrame>() {
    override fun value(current: TextureKeyFrame, enlapsedTime: Long, next: TextureKeyFrame): TextureKeyFrame {
        // questa è un'animazione di tipo discrete, quindi viene preso sempre il valore del frame corente
        return current
    }

    override fun buildFrame(): TextureKeyFrame {
        return TextureKeyFrame()
    }
}